src/syzygy/syzygy.cc

```diff
@@ -1649,13 +1647,17 @@ bool SyzygyTablebase::root_probe(const Position& pos,
       dtz = 1;
     }
     if (result == FAIL) return false;
-    // Better moves are ranked higher. Certain wins are ranked equally.
-    // Losing moves are ranked equally unless a 50-move draw is in sight.
-    int r = dtz > 0
-                ? (dtz + cnt50 <= 99 && !rep ? 1000 : 1000 - (dtz + cnt50))
-                : dtz < 0 ? (-dtz * 2 + cnt50 < 100 ? -1000
-                                                    : -1000 + (-dtz + cnt50))
-                          : 0;
+    int r = 0; //draw by repetition
+    if (next_pos.GetRepetitions() < 2)
```

Comment:
This doesn't work as GetRepetitions is only populated when you push the position on to the history stack, which isn't done here.
This logic change is also wrong - as it loses the concept that if the opponent has repeated you must perform the best option.